### PR TITLE
Deprecate NDefOps and NIso

### PR DESCRIPTION
**Pull request by Villetaneuse**
These files are not used in the stdlib, it's unlikely they are very useful to anyone.

- `Numbers.Natural.Abstract.NDefOps` defines some operations on an abstract Natural domain; the operations should be defined in the corresponding implementations instead.
- `Numbers.Natural.Abstract.NIso` studies isomorphisms between models of NAxioms, this is very rarely (if at all) used.

- [x] Added **changelog**.
- [x] Added / updated **documentation**.

### CODE REVIEW

**Comment by proux01** [comment]
@coqbot run full ci

**Comment by herbelin** [comment]
Indeed, maybe more results for logicians than for a stdlib.

File: doc/changelog/11-standard-library/18668-depr_NDefOps_NIso.rst [hunk 1]
```diff
@@ -0,0 +1,5 @@
+- **Deprecated:**
+  The library files ``Coq.Numbers.Natural.Abstract.NIso``
+  and ``Coq.Numbers.Natural.Abstract.NDefOps`` are deprecated.
```
Comment (proux01) [inline doc/changelog/11-standard-library/18668-depr_NDefOps_NIso.rst:3]: There should be no period here (it's two line below). Could you also take the opportunity to fix all other entries in this `11-standard-library` directory.

**Comment by proux01** [comment]
Apparently, the refman compilation failure was due to the faulty indentation of the changelog of #18564 (cc @andres-erbsen FYI) but only showed up now, when adding something below in the same item list.

**Comment by proux01** [comment]
@coqbot merge now

**Comment by proux01** [comment]
@coqbot merge now

**Review by proux01 — APPROVED** [review]

**Comment by proux01** [comment]
@coqbot merge now

**Comment by Villetaneuse** [comment]
> @coqbot merge now

Thank you very much!

**Pull request merged**